packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -74,20 +75,151 @@ import {
 	channelToDataStore,
 	isDataStoreAliasMessage,
 } from "./dataStore.js";
-import { GCNodeType, detectOutboundRoutesViaDDSKey } from "./gc/index.js";
+import {
+	GCNodeType,
+	detectOutboundRoutesViaDDSKey,
+	trimLeadingAndTrailingSlashes,
+} from "./gc/index.js";
 import {
 	IContainerRuntimeMetadata,
 	nonDataStorePaths,
 	rootHasIsolatedChannels,
 } from "./summary/index.js";
+import { ContainerMessageType, LocalContainerRuntimeMessage } from "./messageTypes.js";
+import { FluidDataStoreRegistry } from "./dataStoreRegistry.js";
+
+/**
+ * Accepted header keys for requests coming to the runtime.
+ * @internal
+ */
+export enum RuntimeHeaders {
+	/** True to wait for a data store to be created and loaded before returning it. */
+	wait = "wait",
+	/** True if the request is coming from an IFluidHandle. */
+	viaHandle = "viaHandle",
+}
+
+/** True if a tombstoned object should be returned without erroring
+ * @alpha
+ */
+export const AllowTombstoneRequestHeaderKey = "allowTombstone"; // Belongs in the enum above, but avoiding the breaking change
+/**
+ * [IRRELEVANT IF throwOnInactiveLoad OPTION NOT SET] True if an inactive object should be returned without erroring
+ * @internal
+ */
+export const AllowInactiveRequestHeaderKey = "allowInactive"; // Belongs in the enum above, but avoiding the breaking change
 
 type PendingAliasResolve = (success: boolean) => void;
 
+interface FluidDataStoreMessage {
+	content: any;
+	type: string;
+}
+
+export function wrapContext(context: IFluidParentContext): IFluidParentContext {
```

Comment:
Can you please add comment on what this is supposed to do. At first glance, it is just creating a copy of the object and if so, why do we even need to do that?